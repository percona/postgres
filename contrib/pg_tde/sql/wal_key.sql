-- basic tests for pg_tde_create_wal_key
-- doesn't test actual wal encryption, as that requires a server restart,
-- only sanity checks for the key creation
CREATE EXTENSION IF NOT EXISTS pg_tde;

SELECT pg_tde_create_wal_key();

SELECT pg_tde_add_global_key_provider_file('file-keyring','/tmp/pg_tde_test_keyring.per');

SELECT pg_tde_create_wal_key();

-- db local principal key with global provider
SELECT pg_tde_set_global_principal_key('test-db-principal-key', 'file-keyring', true);

SELECT pg_tde_create_wal_key();

SELECT pg_tde_set_server_principal_key('test-db-principal-key', 'file-keyring');

-- and now it should work!
SELECT pg_tde_create_wal_key();

-- and now it shouldn't create a new one!
SELECT pg_tde_create_wal_key();

DROP EXTENSION pg_tde;