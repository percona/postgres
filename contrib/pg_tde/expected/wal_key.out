-- basic tests for pg_tde_create_wal_key
-- doesn't test actual wal encryption, as that requires a server restart,
-- only sanity checks for the key creation
CREATE EXTENSION IF NOT EXISTS pg_tde;
NOTICE:  extension "pg_tde" already exists, skipping
SELECT pg_tde_create_wal_key();
ERROR:  failed to retrieve principal key. Create one using pg_tde_set_principal_key before using encrypted tables.
SELECT pg_tde_add_key_provider_file('PG_TDE_GLOBAL', 'file-keyring','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_key_provider_file 
------------------------------
                           -1
(1 row)

SELECT pg_tde_create_wal_key();
ERROR:  failed to retrieve principal key. Create one using pg_tde_set_principal_key before using encrypted tables.
-- db local principal key with global provider
SELECT pg_tde_set_principal_key('test-db-principal-key', 'PG_TDE_GLOBAL', 'file-keyring', true);
ERROR:  failed to create principal key: already exists
SELECT pg_tde_create_wal_key();
ERROR:  failed to retrieve principal key. Create one using pg_tde_set_principal_key before using encrypted tables.
SELECT pg_tde_set_server_principal_key('test-db-principal-key', 'PG_TDE_GLOBAL', 'file-keyring');
 pg_tde_set_server_principal_key 
---------------------------------
 t
(1 row)

-- and now it should work!
SELECT pg_tde_create_wal_key();
 pg_tde_create_wal_key 
-----------------------
 t
(1 row)

-- and now it shouldn't create a new one!
SELECT pg_tde_create_wal_key();
ERROR:  WAL key already exists.
DROP EXTENSION pg_tde;
